posts/views.py

```diff
@@ -1,3 +1,29 @@
-# from django.shortcuts import render
+from rest_framework import viewsets
+from rest_framework.response import Response
 
-# Create your views here.
+from instagram_own_version.permissions import IsAuthor
+from posts.models import Post
+from posts.serializers import PostSerializer
+from users.models import Followers, User
+
+
+class PostViewSet(viewsets.ModelViewSet):
+    queryset = Post.objects.all()
+    permission_classes = (IsAuthor,)
+    serializer_class = PostSerializer
+
+    def list(self, request):
+        queryset = Post.objects.all()
+        user_parameter = int(self.request.GET.get("user_id", None))
+        feed_parameter = self.request.GET.get("feed", None)
+        if user_parameter:
+            user_obj = User.objects.get(id=user_parameter)
+            if feed_parameter:
+                fol = Followers.objects.filter(follower=user_obj).values_list(
+                    "following", flat=True
+                )
+                queryset = queryset.filter(user__in=fol)
+            else:
+                queryset = queryset.filter(user=user_obj)
```

Comment:
Я имею в виду что-то вроде этого. Может где-то ошибся немного, не проверял этот код, но смысл такой. Если есть флажок `user_id` - фильтруем по нему, если есть флажок `feed` - фильтруем по фолловинг. Если есть и то и другое, фильтруем и по тому и другому, нет ни того ни другого - не фильтруем (скорее всего два этих случая не имеют для нас смысла, но не думаю, что стоит их как-то специально обрабатывать)
```suggestion
        try:
            user_id = int(self.request.GET.get("user_id"))
        except (Some errors for cases when user id is non-numerical or None):
            user_id = None
        show_feed = self.request.GET.get("feed") == "true"
        if user_id:
            queryset = queryset.filter(user_id=user_id)
        if show_feed:
            following = Followers.objects.filter(follower=self.request.user).values_list(
                "following", flat=True
            )
            queryset = queryset.filter(user__in=following)
```

Reply:
Done